app/src/main/java/org/mozilla/FirefoxLite/firefoxdownloader/MainActivity.java

```diff
@@ -0,0 +1,294 @@
+    final String URL4 = "http://www.appsapk.com/downloading/latest/Emoji%20Flashlight%20-%20Brightest%20Flashlight%202018-2.0.1.apk";
+    final String URL5 = "http://www.appsapk.com/downloading/latest/Screen%20Recorder-7.7.apk";
+    final String URL6 = "http://www.appsapk.com/downloading/latest/Call%20Recorder%20-%20Automatic%20Call%20Recorder-1.6.0.apk";
+    final String URL7 = "http://www.appsapk.com/downloading/latest/Sound%20Profile%20(+%20volume%20scheduler)-5.25.apk";
+    final String URL8 = "http://www.appsapk.com/downloading/latest/Evernote%20-%20stay%20organized.-7.9.7.apk";
+    final String URL9 = "http://www.appsapk.com/downloading/latest/UC-Browser.apk";
+    final String URL10 = "http://www.appsapk.com/downloading/latest/Barcode%20Scanner-1.2.apk";
+    final String URL11 = "http://download.blender.org/peach/bigbuckbunny_movies/BigBuckBunny_640x360.m4v";
+    final String URL12 = "http://www2.sdfi.edu.cn/netclass/jiaoan/englit/download/Harry%20Potter%20and%20the%20Sorcerer's%20Stone.pdf";
+    final String URL13 = "https://media.giphy.com/media/Bk0CW5frw4qfS/giphy.gif";
+    final String URL14 = "http://techslides.com/demos/sample-videos/small.mp4";
+    final String URL15 = "http://www.sample-videos.com/video/mp4/720/big_buck_bunny_720p_10mb.mp4";
+
+    Button buttonOne;
+    Button buttonCancelOne;
+    Button buttonScheduleOne;
+
+
+
+
+    TextView textViewProgressOne,textViewProgressTwo,textViewProgressThree, textViewProgressFour, textViewProgressFive, textViewProgressSix,
+            textViewProgressSeven, textViewProgressEight, textViewProgressNine,
+            textViewProgressTen, textViewProgressEleven, textViewProgressTwelve,
+            textViewProgressThirteen, textViewProgressFourteen, textViewProgressFifteen;
+
+    ProgressBar progressBarOne, progressBarTwo, progressBarThree,
+            progressBarFour, progressBarFive, progressBarSix,
+            progressBarSeven, progressBarEight, progressBarNine,
+            progressBarTen, progressBarEleven, progressBarTwelve,
+            progressBarThirteen, progressBarFourteen, progressBarFifteen;
+
+    MainActivity mainActivity;
+    private int STORAGE_PERMISSIONS_CODE = 1;
+    private static String dirPath;
+    private  NotificationHelper mNotificationHelper;
+    Long lasttimestamp;
+    MyReceiver myReceiver;
+    String schedulesession;
+
+    private class MyReceiver extends BroadcastReceiver{
+
+        @Override
+        public void onReceive(Context arg0, Intent arg1) {
+            // TODO Auto-generated method stub
+
+            long datapassed = arg1.getLongExtra("DATAPASSED", 0);
+            progressBarOne.setProgress((int)datapassed);
+            String statusofdownload=arg1.getStringExtra("Statusofdownload");
+
+            if (statusofdownload!=null){
+                if(statusofdownload.equals("started")){
+                    progressBarOne.setIndeterminate(false);
```

Comment:
if the user has ten downloading tasks, in this case, the user needs to copy this code ten times?

Here's an example 
https://github.com/mozilla-tw/FirefoxLite/blob/2f0b0c7d13c1fd8f98d8f1b80060591d629ed4e9/app/src/main/java/org/mozilla/focus/fragment/DownloadsFragment.java#L51